SPECS/haproxy-dataplaneapi/haproxy-dataplaneapi.spec

```
@@ -0,0 +1,58 @@
+# https://fedoraproject.org/wiki/PackagingDrafts/Go
+%global commit          284d34bb4aebf36df8ba1ad430adfd72c222d08b
+%global commitdate      20200415
+%global shortcommit     %(c=%{commit}; echo ${c:0:7})
+%global repo            dataplaneapi
+%global cmd             %{repo}
+%global import_path     github.com/haproxytech/%{repo}
+%global build_date      %(date '+%%Y-%%m-%%dT%%H:%%M:%%S')
+# Required to avoid an error once integrating the Go Build ID.
+%global debug_package   %{nil}
+
+Summary:        A sidecar process for managing HAProxy.
+Name:           haproxy-%{repo}
+Version:        1.2.5
+Release:        1%{?dist}
+License:        Apache License 2.0
+URL:            https://%{import_path}
+Group:          Applications/System
+Vendor:         VMware, Inc.
+Distribution:   Photon
+Source0:        https://%{import_path}/archive/%{commit}/%{repo}-%{shortcommit}.tar.gz
+%define sha1    %{repo}=9cfca3f7b240a51bb6e5995350a377aa6d7054f2
+BuildRequires:  go >= 1.13
+Requires:       haproxy >= 2.0.10
+
+%description
+HAProxy Data Plane API is a sidecar process that runs next to HAProxy
+and provides API endpoints for managing HAProxy.
+
+%prep
+%setup -n %{repo}-%{version}
+
+%build
+%define gcflags -N -l
+# The build_id and ldflags_for_build_id are the solution for the issue
+# that the Build ID in Go binaries:
+# - the issue       https://github.com/rpm-software-management/rpm/issues/367
+# - the work-around https://github.com/aws/amazon-ssm-agent/issues/268
+%define build_id %(head -c20 /dev/urandom|od -An -tx1|tr -d '[:space:]')
+%define ldflags_for_build_id -s -w -B 0x%{build_id} -extldflags=-Wl,-z,now,-z,relro,-z,defs
+%define ldflags_for_build_metadata -X main.GitRepo=%{repo} -X main.GitTag=v%{version} -X main.GitCommit=%{commit} -X main.GitDirty= -X main.BuildTime=%{build_date}
+%define ldflags %{ldflags_for_build_id} %{ldflags_for_build_metadata}
+export CGO_ENABLED=0
+go build -gcflags "%{gcflags}" -ldflags "%{ldflags}" -o %{cmd} ./cmd/%{cmd}/
+
+%install
+install -m 755 -D %{cmd} %{buildroot}%{_bindir}/%{cmd}
+
+%clean
+rm -rf %{buildroot}/*
+
+%files
+%defattr(-,root,root)
+%{_bindir}/%{cmd}
```

```suggestion
%{_libexecdir}/%{cmd}
```
`/usr/libexec/haproxy-dataplaneapi` is probably a more appropriate location, since a user wouldn't directly execute the binary, it would only be started by haproxy.